src/Multitenancy.Tracker/TopicSubscriptionTracker.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Cassandra;
+using PushNotifications.Contracts;
+using System.Collections.Generic;
+
+namespace Multitenancy.Tracker
+{
+    public partial class TopicSubscriptionTracker : ITopicSubscriptionTracker
+    {
+        const string CreateTableTemplate = @"CREATE TABLE IF NOT EXISTS ""pushnot_subscriptions"" (cv counter, name varchar, PRIMARY KEY (name));";
+        const string IncrementTemplate = @"UPDATE ""pushnot_subscriptions"" SET cv = cv + 1 WHERE name=?;";
+        const string DecrementTemplate = @"UPDATE ""pushnot_subscriptions"" SET cv = cv - 1 WHERE name=?;";
+        const string GetTemplate = @"SELECT * FROM pushnot_subscriptions WHERE name=?;";
+
+        private readonly ISession _session;
+        private readonly PreparedStatement incrementTemplate;
+        private readonly PreparedStatement decrementTemplate;
+        private readonly PreparedStatement getTemplate;
+
+        public TopicSubscriptionTracker(ISession session)
+        {
+            _session = session;
+
+            _session.Execute(CreateTableTemplate);
```

Comment:
@AlexPlom this will work as long as we deploy the push notifications to a single node OR the cassandra version is >= v4. That said I prefer to have an issue somewhere about schema creation in cassandra. The problem is that cassandra schema mutations are not idempotent and they HAVE TO be synchronized and executed from one process and one thread.

Reply:
This should be resolved with the addition of the RedisLock when we need a dist lock, or the InMemory Lock from Cronus when we run this on a single server.